lib/common/components/forms/custom_date_picker.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_boilerplate/common/components/forms/custom_form_input_class.dart';
+import 'package:intl/intl.dart';
+
+class CustomDatePicker extends StatefulWidget {
+  final CustomFormInput input;
+
+  const CustomDatePicker({Key? key, required this.input}) : super(key: key);
+
+  @override
+  State<CustomDatePicker> createState() => _CustomDatePickerState();
+}
+
+class _CustomDatePickerState extends State<CustomDatePicker> {
+  DateTime selectedDate = DateTime.now();
+
+  Future<void> _selectDate(BuildContext context) async {
+    final DateTime? picked = await showDatePicker(
+      context: context,
+      initialDate: selectedDate,
+      firstDate: DateTime(2015),
```

Comment:
I think this should be earlier like 100+ years from now or/and set a default value which can be changed